maven-plugins/helidon-archetype-maven-plugin/src/main/java/io/helidon/build/maven/archetype/JarMojo.java

```diff
@@ -173,6 +222,79 @@ public void execute() throws MojoExecutionException, MojoFailureException {
         project.getArtifact().setFile(jarFile);
     }
 
+    private void validateArchetypeScripts() throws MojoExecutionException {
+        File schemaFile = getArchetypeSchemaFile();
+        if (schemaFile == null) {
+            return;
+        }
+        try {
+            Validator validator = getValidator(schemaFile);
+            List<Path> xmlFiles = getXmlFiles();
+            DocumentBuilder db = getDocumentBuilder();
+            for (Path xmlFilePath : xmlFiles) {
+                File xmlFile = xmlFilePath.toFile();
+                Document doc = db.parse(xmlFile);
+                if (doc.getDocumentElement().getNodeName().equals(ARCHETYPE_ROOT_ELEMENT)) {
+                    validator.validate(new StreamSource(xmlFile));
+                }
+            }
+        } catch (SAXException | IOException | ParserConfigurationException e) {
+            throw new MojoExecutionException(e.getMessage(), e);
+        }
+    }
+
+    private DocumentBuilder getDocumentBuilder() throws ParserConfigurationException {
+        DocumentBuilderFactory dbf = DocumentBuilderFactory.newInstance();
+        dbf.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+        return dbf.newDocumentBuilder();
+    }
+
+    private Validator getValidator(File schemaFile) throws SAXException {
+        SchemaFactory factory = SchemaFactory.newInstance(SCHEMA_LANG);
+        Schema schema = factory.newSchema(schemaFile);
+        return schema.newValidator();
+    }
+
+    private List<Path> getXmlFiles() throws IOException {
```

Comment:
The method `scanResources` already scans all files to be processed.
You can filter the resources by `.xml` extension and by schema header.